stdlib/n-ary-app.kl

```diff
@@ -105,10 +105,17 @@
                     [(list (arg1 arg2))
                      (pure (list-syntax ('-> arg1 arg2)
                                         stx))]
+                    [(list (arg1))
+                     (pure (list-syntax ('-> arg1)
+                                        stx))]
                     [(cons arg1 args)
                      (pure (list-syntax ('-> arg1
                                           (cons-list-syntax 'my-arrow args stx))
-                                        stx))])]))]))
+                                        stx))]
+                    [()
+                     (pure '(->))])]
+                 [_
+                  (pure '(->))]))]))
```

Comment:
This extends the wrapper to support a syntax which the kernel does not support, namely `->` without parentheses. Hmm, I should probably change `higher-kinded.kl` to test both `->` and `(->)`...

Reply:
Tested in 2847b4e3f2072c9f6c7d24c50bcf3fe8d2fd3812.